src/replication.c

```diff
@@ -1030,28 +1267,81 @@ read_io_resp(spec_t *spec, replica_t *replica, read_event_t *revent)
 	return donecount;
 }
 
+/*
+ * write data on management fd until EAGAIN
+ */
+int
+handle_write_data_event(replica_t *replica)
+{
+	io_event_t wevent;
+	int rc = -1;
+	mgmt_cmd_t *mgmt_cmd;
+
+	MTX_LOCK(&replica->r_mtx);
+	mgmt_cmd = TAILQ_FIRST(&replica->mgmt_cmd_queue);
+	if (!(mgmt_cmd != NULL &&
+		(mgmt_cmd->mgmt_cmd_state == WRITE_IO_SEND_HDR ||
+		mgmt_cmd->mgmt_cmd_state == WRITE_IO_SEND_DATA))) {
+		MTX_UNLOCK(&replica->r_mtx);
+		return rc;
```

Comment:
lets add a comment that we hit this case when this IO in the head of list is waiting for read and in READ related state.